src/HooglePlus/FilterTest.hs

```diff
@@ -168,4 +168,35 @@ handleAnyException = (`catch` (return . handler . show :: SomeException -> IO (M
   where
     handler ex  | "timeout" `isInfixOf` ex = Nothing
                 | otherwise = (Just . Left. UnknownError) ex
-fmtFunction_ = printf "((%s) :: %s) %s" :: String -> String -> String -> String
\ No newline at end of file
+fmtFunction_ = printf "((%s) :: %s) %s" :: String -> String -> String -> String
+
+checkDuplicates :: SampleResult -> Environment -> RType -> UProgram -> IO (Bool, SampleResult)
+checkDuplicates result env goalType prog =
+  checkDuplicates' result modules sigStr body
+  where
+    (modules, sigStr, body, _) = extractSolution env goalType prog
+
+
+checkDuplicates' :: SampleResult -> [String] -> String -> String -> IO (Bool, SampleResult)
+checkDuplicates' result modules sigStr body =
+  case result of
+    EmptySample ->
+      ((`Sample` []) <$> generateInputs modules funcSig) >>= (\r -> checkDuplicates' r modules sigStr body)
+    Sample inputs results -> do
+      evals <- evalResults inputs funcSig body
+      let isDuplicated = evals `elem` results
+      let results' = if isDuplicated then results else evals:results
+
+      let retVal = (not isDuplicated, Sample inputs results')
```

Comment:
`retVal` is generally an unhelpful variable name.
Why do you need to print it anyways? If not, then you don't need this variable at all.